### ISSUE TITLE
Building on Windows

### ISSUE THREAD

**Issue by anko** [anko/eslisp#44]
It has come to my attention that eslisp doesn't build on Windows.

I don't have a Windows machine to test on, but from what I can gather, these are the points of friction, in the form of a todo list:
- [ ] CI-test it, so users on other platforms also know it works. ([Travis doesn't do Windows](https://github.com/travis-ci/travis-ci/issues/216). Alt: [AppVeyor](https://www.appveyor.com/)?)
- [ ] Remove the `make` dependency. (Alt: [Jake](http://howtonode.org/intro-to-jake)? [Gulp](http://gulpjs.com/)?)
- [ ] Some code has file paths fixed to being `/`-delimited. (Should use [`path.join`](https://nodejs.org/api/path.html#path_path_join_path1_path2).)
- [x] Explicitly call `node <filename>` in tests. (Windows doesn't recognise the [shebang line](https://en.wikipedia.org/wiki/Shebang_%28Unix%29). @isiahmeadows' #43 addresses this.)

Anything I've missed? Opinions? Volunteers?

**Comment by vendethiel** [comment]
`make` isn't in issue in itself (it works fine on my machine, though I'd probably prefer a full-js alternative..). The main problem is the incompatible binaries (example in LS, `find`)

**Comment by anko** [comment]
@vendethiel What do you mean by incompatible binaries? Is “LS” [LiveScript](http://livescript.net/)?

**Comment by vendethiel** [comment]
Yes, sorry. [this line in particular](https://github.com/gkz/LiveScript/blob/master/Makefile#L3) breaks building on windows.

**Comment by dead-claudia** [comment]
There's no reason why LiveScript's build process should make _this_ not
buildable on Windows. If that's the case, then it's a massive LiveScript
bug that I'm shocked was never filed _and_ fixed.

I was looking at the fact make doesn't work well on Windows without Cygwin
or the like. I normally use Linux, so it's not an issue for me,  but it
could be an issue for some. Although, you could use npm scripts alone for
this project, because it's simple enough.

And the other issue, that of Windows not understanding the shebang, is
already addressed. And if it's based in Node, the directory separator
shouldn't matter much, even in commands, because Node normalizes those
(i.e. ".\foo" is equivalent to "./foo" on both platforms). And merely using
path.resolve instead of path.join should fix it if you're running shell
commands. (The rest are almost always normalized by Node.)

**Comment by vendethiel** [comment]
No, I just quoted LiveScript's build as an example of something that uses a Makefile but doesn't build on windows.

**Comment by dead-claudia** [comment]
@anko 
1. AppVeyor sounds fine.
2. `npm` would even work. `lsc` works with directories, and you could use `npm run compile` and `{"scripts": {"compile": "lsc src -cbo dest"}}`. The rest should work similarly. (npm scripts are actually executed as shell commands)
3. That _should_ also be fixed by #43, but I haven't tested it on my Windows installation (which is currently somewhat broken and unusually slow - I run a dual boot, spending 99% of my time in Ubuntu).

**Comment by raymond-w-ko** [comment]
Hello, I am looking alternate languages for a project my company is doing. Since I like Lisp, but my company is mostly using Windows, I am interested in getting this working.

What should I be focusing on first?

I guess I should try this out on Linux first, so I can see what is going on, and how the normal build process works.

**Comment by dead-claudia** [comment]
I believe you should already be able to use this on Windows, as the
compiled version on npm should work everywhere. The problem this bug is
discussing is being able to hack on the compiler itself on Windows. You can
still use LiveScript on Windows, even though you can't exactly build the
compiler.
